posts/aws-cognito-and-jhipster-for-the-love-of-oauth-2.0.md

```diff
@@ -0,0 +1,86 @@
+
+An alternative to [Okta](https://www.okta.com/) is to use [AWS Cognito](https://aws.amazon.com/cognito/), especially if your whole architecture is running on AWS. Other features are the possibility to trigger Lambdas on specific events (ex: when a user is confirmed) or to link a user's group with an IAM role. In this blog, I will explain how a JHipster application can be configured to use AWS Cognito instead of Okta/Keycloak. All the working code is accessible at this [GitHub repository](https://github.com/Falydoor/cognito-jhipster).
+
+# Application generation
+
+I decided to generate a standard JHipster application that uses `OAuth 2.0/OIDC` for the authentication. After installing JHipster, you just have to run `jhipster` and use the default values except for the authentication question:
+
+![](https://raw.githubusercontent.com/Falydoor/blog-usa/cognito-jhipster/images/2020/02/cognito-generation.png)
+
+This [.yo-rc.json](https://raw.githubusercontent.com/Falydoor/cognito-jhipster/master/.yo-rc.json) can be used in case you want to generate the same application referenced in this article. The file is placed in the application directory and then running `jhipster` will generate the application without asking any questions.
+
+# User pool creation
+
+After login into the [AWS Management Console](https://console.aws.amazon.com/console/home), browse to the [Cognito service](https://console.aws.amazon.com/cognito/users/?region=us-east-1) and click on `Create a user pool`. A user pool with the default configuration like below is required:
+
+![](https://raw.githubusercontent.com/Falydoor/blog-usa/cognito-jhipster/images/2020/02/cognito-user-pool.png)
+
+## Add users and groups
+
+JHipster uses two groups that must be added to the user pool: `ROLE_USER` and `ROLE_ADMIN`:
+
+![](https://raw.githubusercontent.com/Falydoor/blog-usa/cognito-jhipster/images/2020/02/cognito-roles.png)
+
+After creating the two groups, an `admin` user can be created in order to access the application. Don't forget to assign the user to the two groups!
+
+![](https://raw.githubusercontent.com/Falydoor/blog-usa/cognito-jhipster/images/2020/02/cognito-user.png)
+
+## Create an app client and domain
+
+The last step is to create an app client that will be used by the JHipster app. It is required to allow people authenticate using users created in the user pool. Like before, an app client with the default parameters can be created:
+
+![](https://raw.githubusercontent.com/Falydoor/blog-usa/cognito-jhipster/images/2020/02/cognito-app-client.png)
+
+The app client id and secret are required by the JHipster app, make sure to not share them or push them to a public repository. And also, keep in mind that secrets cannot be changed after an app is created. You can create a new app with a new secret if you want to rotate the secret that you are using.
+
+The callback, sign out URL and the allowed OAuth Flows/Scopes need to have the values below:
```

Comment:
`The callback, sign-out URL, and the allowed`

[hyphenated "sign-out" and added comma after "URL"]